javaslang-benchmark/src/test/java/javaslang/idiom/PatternMatchingBenchmark.java

```diff
@@ -0,0 +1,74 @@
+package javaslang.idiom;
+
+import javaslang.JmhRunner;
+import javaslang.collection.Array;
+import org.junit.Test;
+import org.openjdk.jmh.annotations.*;
+
+import java.util.Random;
+
+import static javaslang.API.*;
+
+/**
+ * Benchmark for nested loops vs javaslang's For().yield comprehensions.
+ *
+ * @see javaslang.API.For2
+ */
+public class PatternMatchingBenchmark {
+    static final Array<Class<?>> CLASSES = Array.of(
+            PatternMatchingMatches.class
+    );
+
+    @Test
+    public void testAsserts() {
+        JmhRunner.runDebugWithAsserts(CLASSES);
+    }
+
+    public static void main(String... args) {
+        JmhRunner.runNormalNoAsserts(CLASSES);
+    }
+
+
+    @State(Scope.Benchmark)
+    public static class Base {
+        int random;
+
+        @Setup
+        public void setup() {
+            random = new Random().nextInt();
+            random = random ^ random; //always 0, but compiler doesn't know
+        }
+    }
+
+    public static class PatternMatchingMatches extends Base {
+
+        @Benchmark
+        public Object javaslang_match_ok() {
+            String res = Match(random).of(
+                    Case($(0), "zero"),
+                    Case($(), "fallback")
+            );
+
+            assert "zero".equals(res);
+            return res;
+        }
+
+        @Benchmark
+        public Object java_switch() {
+            String res;
+
+            switch (random) {
```

Comment:
you're testing binary branching, i.e. an if.
A switch is meant for more branches.
java's `switch` can be linear or can sometimes binary search or a lookup table.
Javaslang's is always linear.

Please test them properly.